sensor/common/collector/service_impl.go

```diff
@@ -0,0 +1,141 @@
+package collector
+
+import (
+	"context"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/grpc/authz/idcheck"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/sensor/common"
+	"github.com/stackrox/rox/sensor/common/message"
+	"google.golang.org/grpc"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// CollectorService is the struct that manages the collector configuration
+type serviceImpl struct {
+	sensor.UnimplementedCollectorServiceServer
+
+	collectorC chan *sensor.MsgToCollector
+
+	connectionManager *connectionManager
+}
+
+func (s *serviceImpl) Notify(e common.SensorComponentEvent) {
+}
```

Comment:
We need to think about the correct behavior of this feature in the Offline mode (i.e., Sensor disconnected from Central) and add here the handling of transition between Online and Offline.

Reply:
What handling is needed in that case?

Reply:
It is feature specific handling and it depends a lot on a scenario. Imagine Sensor looses connection to Central - would you like this feature to pause maybe? Or maybe it should start buffering something hoping that the connection will be back soon? There are many options possible - including doing nothing and continuing normal operation.

Reply:
I don't think anything needs to be done in case sensor looses the connection to central. One could argue that continuing to run as normal wastes resources, but it is such a small amount that I don't think it is worth it.

Reply:
Thanks, that makes sense. How about Sensor that was never connected to Central (due to connection issues) and is transitioning from Offline to Online for the first time? Would we like to do something different in that situation?

Reply:
I don't think so. Do you think anything needs to be done in that case? Sensor should receive the cluster config in that case and it will be passed onto the collectors. We should probably take this conversation to the new PR https://github.com/stackrox/stackrox/pull/12826